### PR TITLE
Improve docs for properties inheritance feature

### PR DESCRIPTION
**Pull request by iturricf**
Comments and suggestions are welcome!

### CODE REVIEW

**Comment by stefanhoth** [comment]
Can one of the admins review this PR?

**Comment by mr-archano** [comment]
@hal9002 retest this please

File: README.md [hunk 1]
```diff
@@ -98,10 +98,46 @@ A problem occurred while evaluating entry:
...
+```gradle
+buildProperties {
+    secrets {
+        file rootProject.file('debug.properties')
+    }
+}
+
+...
+
+    defaultConfig {
```
Comment (ataulm) [inline README.md]: if including the indent, can you also add the `android` closure to show that this config lives there?

File: README.md [hunk 2]
```diff
@@ -98,10 +98,46 @@ A problem occurred while evaluating entry:
...
+        ...
+        buildConfigProperty 'FOO', buildProperties.secrets['foo'] // bar
+        buildConfigProperty 'A_KEY', buildProperties.secrets['aKey'] // overridenPreviousValue
```
Comment (ataulm) [inline README.md]: Typo in `overridenPreviousValue` (doesn't match the value set in `debug.properties`)

File: README.md [hunk 3]
```diff
@@ -98,10 +98,46 @@ A problem occurred while evaluating entry:
 
 #### Properties inheritance
 It might be useful to have properties files that can recursively include
-other properties files (specified via an `include` property).
+other properties files (specified via an `include` property with including file path as value).
```
Comment (ataulm) [inline README.md]: I think the example you've given is really clear and this addition here will not be necessary.

File: README.md [hunk 4]
```diff
@@ -98,10 +98,46 @@ A problem occurred while evaluating entry:
...
 Inherited properties can be overridden by the including set, just redefine
 the property in the file and its value will be used instead of the one
-from the included set.
+from the included set, eg:
```
Comment (ataulm) [inline README.md]: no need for this addition. If you like it, could you change it to:

---
...from the included set.

For example, given a base properties file `base.properties`:

```properties
 foo=bar
 aKey=aValue
 ```

you can override values and add additional ones in another properties file, `debug.properties`:

```properties
 include=/path/to/base.properties
 aKey=overriddenPreviousValue
 aNewKey=aNewValue
 ```

---

File: README.md (hunk 3 above)
Comment (frapontillo) [inline README.md]: Agreeed.

File: README.md (hunk 4 above)
Comment (frapontillo) [inline README.md]: Also, agreed.

File: README.md (hunk 2 above)
Comment (frapontillo) [inline README.md]: 👀

File: README.md [hunk 5]
```diff
@@ -121,22 +121,24 @@ aKey=overriddenPreviousValue
 Then in your `build.gradle`:
 
 ```gradle
-buildProperties {
-    secrets {
-        file rootProject.file('debug.properties')
+android {
+    ...
+    buildProperties {
```
Comment (frapontillo) [inline README.md]: `buildProperties` should go out of the `android` closure.

Comment (ataulm) [inline README.md]:
```gradle
buildProperties {
    secrets {
        file rootProject.file('debug.properties')
    }
}

...

android {
    defaultConfig {
        buildConfigProperty 'FOO', buildProperties.secrets['foo'] // bar
        buildConfigProperty 'A_KEY', buildProperties.secrets['aKey'] // overriddenPreviousValue
        buildConfigProperty 'A_NEW_KEY', buildProperties.secrets['aNewKey'] // aNewValue
    }
}
```

**Comment by iturricf** [comment]
hi @ataulm would you prefer a rebase into one commit?

**Comment by ataulm** [comment]
@iturricf up to you, I have no objections with 100 commits :grin: (but maybe avoid 100 if you can).

**Comment by frapontillo** [comment]
@iturricf I still see the issue of `buildProperties` being in the `android` closure.

**Comment by iturricf** [comment]
oh, I didn't realize `buildProperties` goes outside `android` closure @frapontillo. That's how it looks in my project, indeed, as I've posted in this old issue #14.

nevertheless I'll change it as requested. :+1:

**Comment by iturricf** [comment]
done

**Comment by mr-archano** [comment]
@iturricf if you put it inside the `android` extension you will be able to access only in that scope, that is probably fine in your case. The plugin works independently from the Android plugin and we're trying to keep consistency with the other samples, hence the request for changing the snippet.

**Comment by mr-archano** [comment]
@hal9002 retest this please.

**Comment by frapontillo** [comment]
Thanks @iturricf

**Comment by iturricf** [comment]
@mr-archano cool! good to know. I've added another buildProperties block inside my debug buildType it works great also!

**Comment by iturricf** [comment]
@frapontillo thank you all guys! this was helpful for my project too

**Pull request merged**